core/src/ldml/ldml_transforms.cpp

```diff
@@ -403,25 +403,29 @@ transform_entry::transform_entry(const transform_entry &other)
 
 transform_entry::transform_entry(const std::u32string &from, const std::u32string &to)
     : fFrom(from), fTo(to), fFromPattern(nullptr), fMapFromStrId(), fMapToStrId(), fMapFromList(), fMapToList() {
-  assert(!fFrom.empty()); // TODO-LDML: should not happen?
+  assert(!fFrom.empty());
 
   init();
 }
 
-// TODO-LDML: How do we return errors from here?
 transform_entry::transform_entry(
     const std::u32string &from,
     const std::u32string &to,
     KMX_DWORD mapFrom,
     KMX_DWORD mapTo,
-    const kmx::kmx_plus &kplus)
+    const kmx::kmx_plus &kplus,
+    bool &valid)
     : fFrom(from), fTo(to), fFromPattern(nullptr), fMapFromStrId(mapFrom), fMapToStrId(mapTo) {
+  if (!valid)
+    return; // exit early
```

Comment:
so, if valid came in false we leave with it false.

Only change it from true->false on err, don't set it to true here.